### DISCUSSION TITLE
Create a first conversation with a new contact using the API

### DISCUSSION THREAD

**Discussion by yairShiloach** [chatwoot/chatwoot#2198] [General]
Hi!
We have a app with contacts,
We send a request to api to create this contact in chatwoot and we get a response:
```
{
    "payload": {
        "contact": {
            "additional_attributes": {},
            "availability_status": "offline",
            "email": "info+5@xxxx.coml",
            "id": 17,
            "name": "test",
            "phone_number": "+9758888888",
            "thumbnail": "",
            "custom_attributes": {},
            "contact_inboxes": []
        },
        "contact_inbox": {
            "inbox": {
                "id": 2,
                "channel_id": 1,
                "account_id": 1,
                "name": "inboxname",
                "created_at": "2021-05-01T12:03:56.122Z",
                "updated_at": "2021-05-01T12:03:56.122Z",
                "channel_type": "Channel::TwilioSms",
                "enable_auto_assignment": true,
                "greeting_enabled": false,
                "greeting_message": null,
                "email_address": null,
                "working_hours_enabled": false,
                "out_of_office_message": null,
                "timezone": "UTC"
            },
            "source_id": "412342326-e685-48fb-b571-c3333283eaef"
        }
    }
}
```

Now, we want to start a conversation with this contact and we sent: 
POST: https://www.example.com/api/v1/accounts/1/conversations
```
{
"source_id": "whatsapp:+9758888888"
}
```

but its not working and get a response: 404 not found
```
{
    "error": "Resource could not be found"
}
```

Could not create a new call with a new contact?
Because from chatwoot app it is possible.

**Comment by pucilpet** [comment]
For source_id you need to specify the ID of the contact_inbox, so 412342326-e685-48fb-b571-c3333283eaef in this case.

**Reply by yairShiloach** [reply]
not working:
![Screen Shot 2021-05-02 at 17 16 08](https://user-images.githubusercontent.com/60548009/116816316-2163be00-ab6a-11eb-9638-a185524fcc4b.png)

its work when I set the phone number but not with new contact:
![Screen Shot 2021-05-02 at 17 14 57](https://user-images.githubusercontent.com/60548009/116816353-3e988c80-ab6a-11eb-97e1-a135a0fbdae1.png)

**Comment by sojan-official** [comment]
@yairShiloach  could you show the request payload you used to create the contact.  
for Twilio inboxes, the source IDs should be the phone number.  

In the payload, you have shown 
```
"contact_inbox": {
            "inbox": {
                "id": 2,
                "channel_id": 1,
                "account_id": 1,
                "name": "inboxname",
                "created_at": "2021-05-01T12:03:56.122Z",
                "updated_at": "2021-05-01T12:03:56.122Z",
                "channel_type": "Channel::TwilioSms",
                "enable_auto_assignment": true,
                "greeting_enabled": false,
                "greeting_message": null,
                "email_address": null,
                "working_hours_enabled": false,
                "out_of_office_message": null,
                "timezone": "UTC"
            },
            "source_id": "412342326-e685-48fb-b571-c3333283eaef"
        }
        
 ```
 
 the source id looks invalid.

**Reply by yairShiloach** [reply]
yes
I tried it:
```
{
  "inbox_id": 2,
  "name": "testtt",
  "email": "yai@gmail.com",
  "phone_number": "+972xxxxxxxx"
}
```
And this too:
`whatsapp+972507500636`

If I'm trying to start a conversation with a contact I'd already talked to via twilio with chatwoot - it works.
It does not open the conversation only when I add a new contact, and try to open a new conversation with it that has never been opened before

**Reply by jaderesp** [reply]
Friend, how do I send the message contact's avatar?

**Reply by sojan-official** [reply]
@jaderesp we don't have an API to update the contact avatar at the moment apart from the one available in our JS sdk. The platform will try to fetch the gravatar provided that the contact has an email address

**Comment by sojan-official** [comment]
@yairShiloach  if you are trying to achieve creating a new conversation purely via API  for new contact in case of Twilio inbox. 

1. Create a contact with the correct phone number ( e164 format ) 
2. Create a conversation with the following parameters 
```
{
  inbox_id:  inbox_id,
  contact_id: contact_id,
  source_id: '' //optional ( depending on type of inboxes , not required for twilio )
  message: { content: 'hi' } // optional , it create a new message in conversation 
}
```

the API will automatically identify the required source id or generate a new one in this case.  

3. you can also query `accounts/account_id/contacts/contact_id/contactable_inboxes` to get a list of inboxes with which you can create conversation for a contact 
---

This can also be done via our outbound messages feature available in the CRM, where the same set of APIs are used


<img width="1424" alt="Screenshot 2021-05-04 at 4 04 03 PM" src="https://user-images.githubusercontent.com/73185/116991558-5b5fcc00-acf2-11eb-9845-6c9649a3e2f5.png">


ps: ensure that you are running the latest chatwoot version.

**Reply by yairShiloach** [reply]
Now its work!
thanks for the detailed answer!
